### PR TITLE
Prepare for workflow functional

### PR DESCRIPTION
**Pull request by yokawara**
Implement pipeline for workflow functional testing.
I'm going to modify job names of the feature according to this pipeline structure.
feature file: https://github.com/screwdriver-cd/screwdriver/blob/master/features/workflow.feature

workflow-graph:

## Reference
relates: https://github.com/screwdriver-cd/screwdriver/pull/1328

### CODE REVIEW

File: screwdriver.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            - echo: echo test
+        requires: [SIMPLE]
+    JOIN:
+        steps:
+            - echo: echo test
+        requires: [PARALLEL1, PARALLEL2]
+    FAIL:
+        steps:
+            - echo: exit 1;
+        requires: [~pr, ~commit]
```
Comment (tk3fftk) [inline screwdriver.yaml]: This should be only `~commit`
https://github.com/screwdriver-cd/screwdriver/blob/9e917c4419485e4a844f29533998dc9f6605b337/features/workflow.feature#L22

File: screwdriver.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+shared:
+    image: node:8
+jobs:
+    SIMPLE:
+        steps:
+            - echo: echo test
+        requires: [~pr, ~commit]
```
Comment (tk3fftk) [inline screwdriver.yaml]: Does `SIMPLE` need `~pr`?

File: screwdriver.yaml (hunk 1 above)
Comment (yokawara) [reply screwdriver.yaml]: Thanks, fixed it.

File: screwdriver.yaml (hunk 2 above)
Comment (yokawara) [reply screwdriver.yaml]: I would like to use this at:
https://github.com/screwdriver-cd/screwdriver/blob/master/features/workflow.feature#L118

**Pull request merged**